services/webos-service-remote/message.js

```diff
@@ -0,0 +1,43 @@
+//message.js - a wrapper for the palmbus message type
+
+export class Message {
+    /* Message constructor
+     * takes two arguments, a palmbus message object, and the handle it was received on
+     * third argument is the activityManager instance to use
+     */
+    constructor(message, handle) {
+        this.category = message.category();
+        this.method = message.method();
+        this.isSubscription = message.isSubscription();
+        this.uniqueToken = message.uniqueToken();
+        this.token = message.token();
+        try {
+            this.payload = JSON.parse(message.payload());
+        } catch (e) {
+            // console.error("Message: badly-formatted message payload");
+            // console.error("payload: " + message.payload());
+            this.payload = { badPayload: message.payload() };
+        }
+        this.sender = 'unknown';
+        this.handle = handle;
+    }
+    //* respond to a message, with a JSON-compatible object
+    respond(response) {
+        throw 'Not implemented';
```

Comment:
https://stackoverflow.com/a/11512623

tl;dr throwing strings is not great, use `throw new Error('Not implemented');`

Reply:
Can you also go over the code and update it in a similar manner for all other occurences of throwing raw strings?